ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -92,6 +92,13 @@
   private volatile boolean responseComplete = false;
   // marked as true if force close is required because close() was called.
   private volatile boolean forceClose = false;
+  // the ResponseStatus that will be sent (or has been sent) as a part of the response.
+  private volatile ResponseStatus responseStatus = ResponseStatus.Ok;
+  // the response metadata that was actually sent.
+  private volatile HttpResponse finalResponseMetadata = null;
```

Comment:
I could talk to you about this, but I am only bringing this up so that everyone gets a clear picture of when we want to use `volatile`. I think there is a misconception that variables getting accessed by multiple threads need to be `volatile`.

I am pretty sure that is not the case here as we have discussed this before, but for others's sake, could you elaborate why (by giving a simplistic pseudo code of two threads, if that is not asking too much)?